lib/engine/game/g_18_india/step/special_track.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/special_track'
+
+module Engine
+  module Game
+    module G18India
+      module Step
+        class SpecialTrack < Engine::Step::SpecialTrack
+          # Bypass some Step::Tracker tests for Town to City upgrade: maintain exits, and check new exits are valid
+          # check tile color to active ability
+          def legal_tile_rotation?(entity, hex, tile)
+            return false unless (ability = abilities(entity))
+            return false if tile.color != :yellow && ability.upgrade_count.zero?
+            return false if tile.color == :yellow && ability.lay_count.zero?
+
+            old_tile = hex.tile
+            if @game.yellow_town_to_city_upgrade?(old_tile, tile)
+              all_new_exits_valid = tile.exits.all? { |edge| hex.neighbors[edge] }
+              return false unless all_new_exits_valid
+
+              return old_tile.paths.all? { |old| tile.paths.any? { |new| old.exits == new.exits } } &&
+                     !(tile.exits & hex_neighbors(entity, hex)).empty?
```

Comment:
What are you testing here? I can't follow the nested iterations.

Reply:
This section is for a yellow town to green single city upgrade.  This test is to ensure the old exits are preserved on the new tile.  

I have copied this from another game:
https://github.com/tobymao/18xx/blob/02f8ebe84a373e5768337512c2de0a45f1fede76/lib/engine/game/g_18_ny/step/track.rb#L23

Reply:
That seems a very complicated way to compare whether the exits match.
Would `(old_tile.exits - tile.exits).empty?` work?

Reply:
Ah, I see what the original test is doing. It is for track paths that run from edge to edge, the test makes sure that these are preserved. These tiles only have paths between edges and the central town/city, so the simpler test should be enough.

Reply:
I have verified that the simpler test works.  Thanks!